src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.cpp

```diff
@@ -35,6 +35,17 @@ using namespace mlir;
 namespace onnx_mlir {
 namespace tosa {
 
+int64_t convertNegativeAxis(int64_t axis, int64_t inputRank) {
+  if (axis < 0)
+    axis += inputRank;
+
+  // Check if axis is in correct range.
+  assert(
+      (axis >= 0 && axis < inputRank) && "axis attribute not in correct range");
+
+  return axis;
+}
+
```

Comment:
Its ok to add that code, but note that there is already a version of this in `ShapeHelper.hpp`
```
// Check if axis is in [-rank, rank), or [-rank, rank] when includeRank is
// true.  Assert when not in range. Return positive axis.
int64_t getAxisInRange(int64_t axis, int64_t rank, bool includeRank = false);
int64_t getAxisInRange(int64_t axis, mlir::Value val, bool includeRank = false);
```